File: VZ02/src/Cell.java
import util.*;

/**
 * @author      Irfan Ariq <13515112@std.stei.itb.ac.id>
 * @version     1.0
 * @since       1.0
 */
public class Cell {
    /**
     * type define what kind of cell it is
     */
    private char type;
    /**
     * hewan is a pointer to an Animal object that
     * represent an animal that exist in a cell
     */
    private Animal hewan;
    /**
     * Constructor
     * @param newType type of cell
     */
    Cell(char newType) {
        type = newType;
        hewan = null;
    }
    /**
     * Get type of cell
     * @return return the type of cell
     */
    public char getType() {
        return type;
    }
    /**
     * Get pointer to Animal that exist in the cell
     * @return return the pointer to Animal
     */
    public Animal getAnimal() {
        return hewan;
    }
    /**
     * Set a new type for the cell
     * @param new_type : as new type that is going to be assigned
     */
    protected void setType(char new_type) {
        type = new_type;
    }
    /**
     * Set a new pointer to Animal for the cell
     * @param A : as new pointer to Animal that is going to be assigned
     */
    protected void setAnimal(Animal A) {
        hewan = A;
    }
    /**
     * Validate a Land type cell
     * @return True if cell type is Land, false if not
     */
    public boolean isLH() {
        return (type == 'L');
    }
    /**
     * Validate a Water type cell
     * @return True if cell type is Water, false if not
     */
    public boolean isWH() {
        return (type == 'W');
    }
    /**
     * Validate an Air type cell
     * @return True if cell type is Air, false if not
     */
    public boolean isAH() {
        return (type == 'A');
    }
    /**
     * Validate a Restaurant type cell
     * @return True if cell type is Restaurant, false if not
     */
    public boolean isRestaurant() {
        return (type == 'R');
    }
    /**
     * Validate a Park type cell
     * @return True if cell type is Park, false if not
     */
    public boolean isPark() {
        return (type == 'P');
    }
    /**
     * Validate an Entrance type cell
     * @return True if cell type is Entrance, false if not
     */
    public boolean isEntrance() {
        return (type == 'E');
    }
    /**
     * Validate an Exit type cell
     * @return True if cell type is Exit, false if not
     */
    public boolean isExit() {
        return (type == 'e');
    }
    /**
     * Validate a Road type cell
     * @return True if cell type is Road, false if not
     */
    public boolean isRoad() {
        return (type == 'r' || isEntrance() || isExit());
    }
    /**
     * Validate whether there is an Animal in the cell
     * @return True if Animal exist, false if not
     */
    public boolean isAnimal() {
        return (hewan != null);
    }
    /**
     * Display Cell into monitor
     * <p>
     * Display cell based on its characteristics
     */
    public void render(){
        if(isAnimal()) {
            Animal temp;
            temp = getAnimal();
            temp.render();
        }else {
            switch (type) {
                case 'L':
                    System.out.print(Global.ANSI_PURPLE_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'W':
                    System.out.print(Global.ANSI_BLUE_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'A':
                    System.out.print(Global.ANSI_YELLOW_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'R':
                    System.out.print(Global.ANSI_PURPLE_BACKGROUND+ "  " + Global.ANSI_RESET);
                    break;
                case 'r':
                    System.out.print(Global.ANSI_BLACK_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'E':
                    System.out.print(Global.ANSI_BLACK_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'e':
                    System.out.print(Global.ANSI_BLACK_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
                case 'P':
                    System.out.print(Global.ANSI_GREEN_BACKGROUND + "  " + Global.ANSI_RESET);
                    break;
            }
        }
    }
}

File: C++/VZ03/src/cell.cpp
/*
 * author : Dery (13515097)
 */

#include "../include/cell.h"
#include "../include/animal.h"

/*
  * @brief fungsi getType
  * berfungsi mengembalikan type di cell tersebut
  */
char Cell::getType(){
    return type;
}
/*
  * @brief fungsi getAnimal
  * berfungsi mengembalikan animal di cell tersebut
  * jika ada
  */
Animal* Cell::getAnimal(){
    return hewan;
}
/*
  * @brief fungsi setType
  * berfungsi mengassign type cell tersebut
  * @param _type : type yg akan di assign
  */
void Cell::setType(char _type){
    type = _type;
}
/*
  * @brief fungsi setAnimal
  * berfungsi mengassign seekor Animal di cell tersebut
  * @param A : Animal yg akan di assign
  */
void Cell::setAnimal(Animal * A){
    hewan = A;

}
/*
  * @brief fungsi isLH
  * berfungsi mengembalikan apakah cell tersebut adalah
  * LandHabitat atau bukan
  */
bool Cell::isLH(){
    return (type == 'L');
}
/*
  * @brief fungsi isWH
  * berfungsi mengembalikan apakah cell tersebut adalah
  * WaterHabitat atau bukan
  */
bool Cell::isWH(){
    return (type == 'W');
}
/*
  * @brief fungsi isAH
  * berfungsi mengembalikan apakah cell tersebut adalah
  * AirHabitat atau bukan
  */
bool Cell::isAH(){
    return (type == 'A');
}
/*
  * @brief fungsi isRestaurant
  * berfungsi mengembalikan apakah cell tersebut adalah
  * Restaurant atau bukan
  */
bool Cell::isRestaurant(){
    return (type == 'R');
}
/*
  * @brief fungsi isRoad
  * berfungsi mengembalikan apakah cell tersebut adalah
  * Road atau bukan
  */
bool Cell::isRoad(){
    return (type == 'r');
}
/*
  * @brief fungsi isPark
  * berfungsi mengembalikan apakah cell tersebut adalah
  * Park atau bukan
  */
bool Cell::isPark(){
    return (type == 'P');
}
/*
  * @brief fungsi isEntrance
  * berfungsi mengembalikan apakah cell tersebut adalah
  * Entrance atau bukan
  */
bool Cell::isEntrance(){
    return (type == 'E');
}
/*
  * @brief fungsi isExit
  * berfungsi mengembalikan apakah cell tersebut adalah
  * Exit atau bukan
  */
bool Cell::isExit(){
    return (type == 'e');
}
/*
  * @brief fungsi isAnimal
  * berfungsi mengembalikan apakah cell tersebut sedang
  * ditempati oleh seekor animal atau tidak
  */
bool Cell::isAnimal(){
    return (hewan != NULL);
}
